### PR TITLE
hypervisor: Pull and build iced-x86 only for mshv

### PR DESCRIPTION
**Pull request by sameo**
And bump iced-x86 to 1.10.0

### CODE REVIEW

File: hypervisor/src/arch/mod.rs [hunk 1]
```diff
@@ -11,6 +11,7 @@
 // Copyright © 2020, Microsoft Corporation
 //
 
+#[cfg(all(feature = "mshv"))]
```
Comment (liuw) [inline hypervisor/src/arch/mod.rs:14]: `#[cfg(feature="mshv")]` should be good enough because there is only one feature specified, right?

**Review by liuw — COMMENTED** [review]
Thanks. I had a patch similar to patch 1 but fell short of posting it because that would mean the x86 instruction emulator will have no more testing from upstream.

**Comment by sameo** [comment]
> Thanks. I had a patch similar to patch 1 but fell short of posting it because that would mean the x86 instruction emulator will have no more testing from upstream.

For some reasons I still believed that we were doing unit testing through gh actions. We're not...I think we should not restrict unit testing to kvm. Let me look into that.

**Comment by liuw** [comment]
> 
> 
> > Thanks. I had a patch similar to patch 1 but fell short of posting it because that would mean the x86 instruction emulator will have no more testing from upstream.
> 
> For some reasons I still believed that we were still doing unit testing through gh actions. We're not...I think we should not restrict unit testing to kvm. Let me look into that.

My line of thinking was briefly outlined in https://github.com/cloud-hypervisor/cloud-hypervisor/issues/2048#issuecomment-741919064 .

Maybe there is another way. We can perhaps supply a command like `cargo test -p hypervisor blah...` to test the emulator only.

**Comment by sameo** [comment]
Replaced with #2068

**Pull request closed without merging**